plugins/woocommerce/includes/shortcodes/class-wc-shortcode-checkout.php

```diff
@@ -195,11 +195,8 @@ private static function order_pay( $order_id ) {
 				);
 				WC()->customer->save();
 
-				$available_gateways = WC()->payment_gateways->get_available_payment_gateways();
-
-				if ( count( $available_gateways ) ) {
```

Comment:
What if there are no payment gateways available, this change will break no?

Also, I'd like to get the product's opinion on this if this is the expected behavior and if this is something we'd like to change.

cc @jamesckemp

Reply:
Method I copied from ```plugins/woocommerce/includes/wc-template-functions.php```:
```php
function woocommerce_checkout_payment() {
		if ( WC()->cart->needs_payment() ) {
			$available_gateways = WC()->payment_gateways()->get_available_payment_gateways();
			WC()->payment_gateways()->set_current_gateway( $available_gateways );
		} else {
			$available_gateways = array();
		}

		/// ...
	}
```
It should work the same in my opinion.